src/checkbox/checkbox.js

```diff
@@ -15,255 +15,214 @@
+      IS_UPGRADED: 'is-upgraded'
+    };
+  }
 
   /**
-   * Handle focus of element.
+   * Initialize all self-managed instances under the given node.
+   * This will only initialize components that have specifically been marked
+   * for self-management.
    *
-   * @param {Event} event The event that fired.
-   * @private
-   */
-  MaterialCheckbox.prototype.onFocus_ = function(event) {
-    this.element_.classList.add(this.CssClasses_.IS_FOCUSED);
-  };
+   * @param {Node} docRoot The node under which to look for components.
+   * @export
+   */
+  static initComponents(docRoot) {
+    var nodes = docRoot.querySelectorAll(
+        `.${MaterialCheckbox.classes_.ROOT}.${MaterialCheckbox.classes_.JS}`);
+    for (let i = 0; i < nodes.length; i++) {
+      // Attach new component to DOM property.
+      nodes[i][MaterialCheckbox.strings_.CLASS_NAME] =
+          new MaterialCheckbox(nodes[i]);
+    }
+  }
 
   /**
-   * Handle lost focus of element.
+   * Attach all listeners to the DOM.
    *
-   * @param {Event} event The event that fired.
-   * @private
+   * @export
    */
-  MaterialCheckbox.prototype.onBlur_ = function(event) {
-    this.element_.classList.remove(this.CssClasses_.IS_FOCUSED);
-  };
+  addEventListeners() {
+    this.input_.addEventListener('change', this.changeListener_);
+    this.input_.addEventListener('focus', this.focusListener_);
+    this.input_.addEventListener('blur', this.blurListener_);
+    this.root_.addEventListener('mouseup', this.mouseUpListener_);
+  }
 
   /**
-   * Handle mouseup.
+   * Remove all listeners from the DOM.
    *
-   * @param {Event} event The event that fired.
-   * @private
+   * @export
    */
-  MaterialCheckbox.prototype.onMouseUp_ = function(event) {
-    this.blur_();
-  };
+  removeEventListeners() {
+    this.input_.removeEventListener('change', this.changeListener_);
+    this.input_.removeEventListener('focus', this.focusListener_);
+    this.input_.removeEventListener('blur', this.blurListener_);
+    this.root_.removeEventListener('mouseup', this.mouseUpListener_);
+  }
 
   /**
-   * Handle class updates.
+   * Return class name as a string. Useful for automation after obfuscation.
    *
-   * @private
+   * @export
    */
-  MaterialCheckbox.prototype.updateClasses_ = function() {
-    this.checkDisabled();
-    this.checkToggleState();
-  };
+  get classAsString() {
+    return MaterialCheckbox.strings_.CLASS_NAME;
+  }
 
   /**
-   * Add blur.
+   * Set "checked" value on checkbox.
    *
-   * @private
+   * @param {boolean} value The value to set the property to.
+   * @export
    */
-  MaterialCheckbox.prototype.blur_ = function() {
-    // TODO: figure out why there's a focus event being fired after our blur,
-    // so that we can avoid this hack.
-    window.setTimeout(function() {
-      this.inputElement_.blur();
-    }.bind(this), /** @type {number} */ (this.Constant_.TINY_TIMEOUT));
-  };
-
-  // Public methods.
+  set checked(value) {
```

Comment:
The best part is that getters and setters are actually ES5, so we have absolutely no problem using them in our public API :)